browser/ui/webui/brave_shields/shields_panel_data_handler.cc

```diff
@@ -52,79 +51,75 @@ void ShieldsPanelDataHandler::GetSiteBlockInfo(
 
 void ShieldsPanelDataHandler::GetSiteSettings(
     GetSiteSettingsCallback callback) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (!active_shields_data_controller_)
+    return;
 
   SiteSettings settings;
-  settings.ad_block_mode = shields_data_ctrlr->GetAdBlockMode();
-  settings.fingerprint_mode = shields_data_ctrlr->GetFingerprintMode();
-  settings.cookie_block_mode = shields_data_ctrlr->GetCookieBlockMode();
+  settings.ad_block_mode = active_shields_data_controller_->GetAdBlockMode();
+  settings.fingerprint_mode =
+      active_shields_data_controller_->GetFingerprintMode();
+  settings.cookie_block_mode =
+      active_shields_data_controller_->GetCookieBlockMode();
   settings.is_https_everywhere_enabled =
-      shields_data_ctrlr->GetHTTPSEverywhereEnabled();
-  settings.is_noscript_enabled = shields_data_ctrlr->GetNoScriptEnabled();
+      active_shields_data_controller_->GetHTTPSEverywhereEnabled();
+  settings.is_noscript_enabled =
+      active_shields_data_controller_->GetNoScriptEnabled();
 
   std::move(callback).Run(settings.Clone());
 }
 
 void ShieldsPanelDataHandler::SetAdBlockMode(AdBlockMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  if (active_shields_data_controller_)
```

Comment:
shouldn't this logic be: `if (!active_shields_data_controller_)` ? if so, all below needs to be changed

Reply:
our tests should fail here but all CI shows green. we might want to add tests for the PanelDataHandler here as a follow-up